controllers/feeds/getFeeds.js

```diff
@@ -102,7 +102,11 @@ module.exports = async (req, res) => {
 
           // TODO: PLEASE ENABLE THIS CHECKER AFTER SCORING HAS BEEN FIXED
           if (now < dateExpired || item.duration_feed == 'never') {
-            let newItem = await modifyAnonimityPost(item);
+            const isBlurredPost = await UserFollowUserFunction.checkIsBlurredPost(
+              UserFollowUser,
+              req.userId
+            );
+            let newItem = await modifyAnonimityPost(item, isBlurredPost);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The asynchronous operation `UserFollowUserFunction.checkIsBlurredPost(UserFollowUser, req.userId)` is being awaited inside a loop. This can lead to performance issues because each iteration will wait for the promise to resolve before moving on to the next one. Consider refactoring this to use `Promise.all()` to allow these operations to run concurrently. 

Also, the `req.userId` is passed as an argument to `checkIsBlurredPost()`. It seems like you're checking if the current user's post should be blurred, but the PR description says the blur depends on the number of followers of the post's author. Make sure to pass the correct userId to the function.

Here's a potential fix:

```diff
let checkBlurPromises = [];
// Inside your loop
checkBlurPromises.push(UserFollowUserFunction.checkIsBlurredPost(UserFollowUser, item.actor.id)); // Assuming item.actor.id is the author's id

// After your loop
let blurResults = await Promise.all(checkBlurPromises);
// Now you can use blurResults array to get the result for each post
```
Please adjust the above snippet according to your actual data structure and flow.


<!-- This is an auto-generated comment by OSS CodeRabbit -->